### ISSUE TITLE
Fatal Exception: java.lang.ArrayIndexOutOfBoundsException length=15; index=15

### ISSUE THREAD

**Issue by yeyintkoko** [uxcam/react-native-ux-cam#29]
It is crashing our app in uxcam.jar file in react-native-ux-cam (v5.3.3). Here is the firebase Crashlytics log:

```
Fatal Exception: java.lang.ArrayIndexOutOfBoundsException: length=15; index=15
       at java.util.ArrayList.add(ArrayList.java:468)
       at org.json.JSONArray.put(JSONArray.java:185)
       at com.uxcam.internals.cf.a(:147)
       at com.uxcam.internals.cf.a(:12)
       at com.uxcam.internals.bl.a(:8)
       at com.uxcam.internals.ay$1.onResponse(:300)
       at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)
```

May be you are indexing out of the array's length in the lib somewhere?

`ArrayIndexOutOfBoundsException length=15; index=15`

Could you please fix and update the uxcam RN package asap?
Thanks.

**Comment by fredikey** [comment]
Caught the same problem in v5.3.3

```
java.lang.ArrayIndexOutOfBoundsException: length=3; index=-1
    at com.uxcam.internals.bm.a
    at com.uxcam.internals.cf.a
    at com.uxcam.internals.cf.a
    at com.uxcam.internals.bl.a
    at com.uxcam.internals.ay$1.onFailure
    at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:525)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    at java.lang.Thread.run(Thread.java:929)

**Comment by hariomsharma-jmd** [comment]
Same problem on v5.3.3

java.lang.ArrayIndexOutOfBoundsException: length=15; index=15
    at java.util.ArrayList.add(ArrayList.java:468)
    at org.json.JSONArray.put(JSONArray.java:187)
    at com.uxcam.internals.cf.a
    at com.uxcam.internals.cf.a
    at com.uxcam.internals.bl.a
    at com.uxcam.internals.ay$1.onResponse
    at okhttp3.RealCall$AsyncCall.execute(RealCall.java:206)
    at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    at java.lang.Thread.run(Thread.java:923)

Any expected timeline to fix this issue?

**Comment by DFDIEGOH** [comment]
Hi everyone! We are planning on releasing a new version of our React Native plugin with a fix to this issue. 

Saying this, we plan on releasing this really soon but feel free to reach out in case you have further questions on this regard.

**Comment by DFDIEGOH** [comment]
Hi @yeyintkoko / @hariomsharma-jmd / @fredikey -

Just wanted to let you know that we have released a new version of our React Native plugin, this is version 5.3.4.

Can you please upgrade it and check that everything is working just fine on your end?

**Comment by yeyintkoko** [comment]
v5.3.4 seems to fix this issue.

**Comment by bharatsatya** [comment]
Hi @yeyintkoko 

I have a similar code in our logger and the app crashes with same exception, what fix you added to handle this ? 

The crash i am getting

java.lang.ArrayIndexOutOfBoundsException: length=244; index=244\\n\\tat java.util.ArrayList.add(ArrayList.java:468)\\n\\tat org.json.JSONArray.put(JSONArray.java:185)

**Comment by yeyintkoko** [comment]
> Hi @yeyintkoko
> 
> I have a similar code in our logger and the app crashes with same exception, what fix you added to handle this ?
> 
> The crash i am getting
> 
> java.lang.ArrayIndexOutOfBoundsException: length=244; index=244\n\tat java.util.ArrayList.add(ArrayList.java:468)\n\tat org.json.JSONArray.put(JSONArray.java:185)

It was fixed in v5.3.4

**Comment by bharatsatya** [comment]
Hi @yeyintkoko 

I got that, since we also see same issue with another library, was more interested in how you fixed the issue ? We are not using react-native-ux-cam
